src/store.rs

```diff
@@ -216,53 +64,49 @@ pub trait Store: Clone + Send + Sync {
         Ok(ipld.clone())
     }
 
-    /// Recursively gets a block and all it's references, inserts them into the store,
-    /// pins the root and unpins the old root.
-    ///
-    /// If a block wasn't found it returns a `BlockNotFound` error without modifying the store.
-    async fn sync<'a, I: Into<Cow<'a, Cid>> + Send, N: Into<Cow<'a, Cid>> + Send>(
-        &self,
-        old: Option<I>,
-        new: N,
-    ) -> Result<()>
-    where
-        Ipld: Decode<<Self::Params as StoreParams>::Codecs>,
-    {
-        let mut visited = HashSet::new();
-        let new: Cow<'_, Cid> = new.into();
-        let mut tx = Transaction::new();
-        let mut stack: Vec<Cid> = vec![new.deref().clone()];
-        while let Some(cid) = stack.pop() {
-            if visited.contains(&cid) {
-                continue;
-            }
-            let block = self.get(&cid).await?;
-            for r in block.references()? {
-                stack.push(r);
-            }
-            tx.import(block)?;
-            visited.insert(cid);
-        }
-        tx.update(old, new);
-        self.commit(tx).await
-    }
+    /// Creates an alias for a `Cid`. To alias a block all it's recursive references
+    /// must be in the store. If blocks are missing, they will be fetched from the network. If
+    /// they aren't found, it will return a `BlockNotFound` error.
```

Comment:
If the blocks are fetched over the network, a call to this method could take a long time. So when exactly do you return the `BlockNotFound` error? After some configurable timeout?

Reply:
Coming up with good defaults/heuristics for this is going to be hard. E.g. you create an alias to a single block without links that is just not there, stopping after 2 minutes might be fine. You create an alias to some huge tree, it will take an hour of network traffic to get all the data. So ideally you would stop this after some period of "no progress".

Nevertheless, fine for now. I just anticipate that this part is going to get more complex.

Reply:
every fetch has is bounded. in ipfs-embed you can configure it, currently I'm using a 3s by default. so on the first block that times out the operation gets aborted